test/controllers/CashAccountPaymentSearchControllerSpec.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.AppConfig
+import connectors.*
+import models.response.*
+import models.*
+import org.mockito.ArgumentMatchers.{any, eq => eqTo}
+import org.mockito.Mockito
+import org.mockito.Mockito.when
+import play.api.Application
+import play.api.i18n.Messages
+import play.api.inject.bind
+import play.api.test.FakeRequest
+import play.api.test.Helpers.*
+import repositories.CashAccountSearchRepository
+import utils.SpecBase
+import utils.TestData.{PAYMENT_SEARCH_VALUE, SEQ_PAYMENT_DETAILS_CONTAINER_01}
+
+import scala.concurrent.Future
+
+class CashAccountPaymentSearchControllerSpec extends SpecBase {
+
+  "search" must {
+
+    "return OK view when search results are available in cache repository" in new Setup {
+
+      val cashAccountTransactionSearchResponseDetail: CashAccountTransactionSearchResponseDetail =
+        CashAccountTransactionSearchResponseDetail(can = cashAccountNumber, eoriDetails = Seq.empty,
+          declarations = None, paymentsWithdrawalsAndTransfers = Some(SEQ_PAYMENT_DETAILS_CONTAINER_01))
+
+      when(mockCustomsFinancialsApiConnector.getCashAccount(eqTo(eori))(any, any))
+        .thenReturn(Future.successful(Some(cashAccount)))
+
+      when(mockCashAccountSearchRepo.get(any[String]))
+        .thenReturn(Future.successful(Some(cashAccountTransactionSearchResponseDetail)))
+
+      running(app) {
+        val request =
+          FakeRequest(GET, routes.CashAccountPaymentSearchController.search(PAYMENT_SEARCH_VALUE, Some(1)).url)
+            .withSession("eori" -> eori)
+
+        val result = route(app, request).value
+        status(result) mustEqual OK
+      }
+    }
+
+    "return NOT Found view when search results are not available in cache repository" in new Setup {
```

Comment:
Could you please add a test case for below values? Also update NOT _FOUND to the correct view as mentioned in earlier comments

when value is successfully retrieved but paymentTransfersList.paymentsWithdrawalsAndTransfers is None